File: app/schemas/job.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel

from app.models.job import JobStatus, JobType

class JobBase(BaseModel):
    name: str
    description: Optional[str] = None
    job_type: JobType = JobType.FETCH_LEADS
    campaign_id: Optional[str] = None

class JobCreate(JobBase):
    pass

class JobInDB(JobBase):
    id: int
    task_id: Optional[str] = None
    job_type: JobType
    status: JobStatus
    result: Optional[str] = None
    error: Optional[str] = None
    campaign_id: Optional[str] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class JobResponse(JobInDB):
    pass

File: frontend/src/types/job.ts
// Job Types based on backend schemas

export enum JobStatus {
  PENDING = 'pending',
  PROCESSING = 'processing',
  COMPLETED = 'completed',
  FAILED = 'failed',
  CANCELLED = 'cancelled',
  PAUSED = 'paused'
}

export enum JobType {
  FETCH_LEADS = 'FETCH_LEADS',
  ENRICH_LEAD = 'ENRICH_LEAD',
  CLEANUP_CAMPAIGN = 'CLEANUP_CAMPAIGN'
}

export interface JobBase {
  name: string;
  description?: string;
}

export interface JobCreate extends JobBase {}

export interface JobUpdate {
  status?: JobStatus;
  result?: string;
  error?: string;
}

export interface JobResponse extends JobBase {
  id: number;
  task_id: string;
  job_type: JobType;
  status: JobStatus;
  result?: string;
  error?: string;
  campaign_id?: string;
  created_at: string;
  updated_at?: string;
  completed_at?: string;
}

// Filter interfaces for API queries
export interface JobListFilters {
  skip?: number;
  limit?: number;
  status?: JobStatus;
  campaign_id?: string;
}

// API Response interfaces
export interface JobListResponse {
  jobs: JobResponse[];
  total?: number;
} 